src/types.jl

```diff
@@ -97,13 +52,416 @@ for Model in [:NeuralNetworkClassifier, :ImageClassifier]
+- `predict_mode(mach, Xnew)`: Return the modes of the probabilistic predictions
+  returned above.
+
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `chain`: The trained "chain", or series of layers, functions, and activations which make up the neural network.
+
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_losses`: The history of training losses, a vector containing the history of all the losses during training. The first element of the vector is the initial penalized loss. After the first element, the nth element corresponds to the loss of epoch n-1.
+
+# Examples
+
+In this example we build a classification model using the Iris dataset.
+```julia
+using MLJ
+using Flux
+import RDatasets
+
+using Random
+Random.seed!(123)
+
+MLJ.color_off()
+
+using Plots
+pyplot(size=(600, 300*(sqrt(5)-1)));
+```
+This is a very basic example, using a default builder and no standardization.
+For a more advance illustration, see [`NeuralNetworkRegressor`](@ref) or [`ImageClassifier`](@ref). First, we can load the data:
+```julia
+iris = RDatasets.dataset("datasets", "iris");
+y, X = unpack(iris, ==(:Species), colname -> true, rng=123);
+NeuralNetworkClassifier = @load NeuralNetworkClassifier
+clf = NeuralNetworkClassifier()
+```
+Next, we can train the model:
+```julia
+import Random.seed!; seed!(123)
+mach = machine(clf, X, y)
+fit!(mach)
+```
+We can train the model in an incremental fashion with the `optimizer_changes_trigger_retraining` flag set to false (which is by default). Here, we change the number of iterations and the learning rate of the optimiser:
+```julia
+clf.optimiser.eta = clf.optimiser.eta * 2
+clf.epochs = clf.epochs + 5
+
+# note that if the optimizer_changes_trigger_retraining flag was set to true
+# the model would be completely retrained from scratch because the optimizer was
+# updated
+fit!(mach, verbosity=2);
+```
+We can inspect the mean training loss using the `cross_entropy` function:
+```julia
+
+training_loss = cross_entropy(predict(mach, X), y) |> mean
+
+```
+And we can access the Flux chain (model) using `fitted_params`:
+```julia
+chain = fitted_params(mach).chain
+```
+Finally, we can see how the out-of-sample performance changes over time, using the `learning_curve` function
+```julia
+r = range(clf, :epochs, lower=1, upper=200, scale=:log10)
+curve = learning_curve(clf, X, y,
+                     range=r,
+                     resampling=Holdout(fraction_train=0.7),
+                     measure=cross_entropy)
+using Plots
+plot(curve.parameter_values,
+     curve.measurements,
+     xlab=curve.parameter_name,
+     xscale=curve.parameter_scale,
+     ylab = "Cross Entropy")
+
+savefig("iris_history.png")
+```
+See also
+[`ImageClassifier`](@ref)
+"""
+NeuralNetworkClassifier
+
+"""
+$(MMI.doc_header(ImageClassifier))
+
+`ImageClassifier`: A neural network model for making probabilistic
+"predictions of a `GrayImage` target, given a table of `Continuous` features.
```

Comment:
There are no tables here 😄 .
```suggestion
`ImageClassifier` classifies images using a neural network adapted to the type
 of images provided (color or greyscale). Predictions are probabistic. Users 
 provide a recipe for constructing the network, based on properties of the image 
 encountered, by specifying an appropriate `builder`. See MLJFlux documentation 
 for more on builders. 
```

(Note that we will be amending this as soon as #208 is merged, and possibly before this current PR is merged.)